File: AutomationService/AutomationService/Features/MailboxMonitor/Domain/Services/ReceivedReply.cs
using AutomationService.Features.ZentralrufAutomation.Domain.Services;

namespace AutomationService.Features.MailboxMonitor.Domain.Services;

/// <summary>
/// Eine vom Postfach-Monitor erfasste und geparste Zentralruf-Antwort, wie der
/// Store (<see cref="IReceivedReplyStore"/>) sie liefert. Die Treffer werden in
/// der SQLite-Datenbank gehalten (überdauern den Neustart); die Oberfläche ruft
/// sie ab und quittiert sie. Best-effort wird die Antwort über die Referenz mit
/// einem Vorgang verknüpft — das Übernehmen bleibt der bestätigte Frontend-Schritt.
/// </summary>
public sealed record ReceivedReply
{
    /// <summary>Stabile, URL-sichere ID für Abruf und Quittung über die API.</summary>
    public required string Id { get; init; }

    /// <summary>Zeitpunkt der Erfassung durch den Monitor.</summary>
    public required DateTimeOffset ReceivedAt { get; init; }

    public required string? Subject { get; init; }

    public required string? From { get; init; }

    /// <summary>Die aus der Mail extrahierten Antwortdaten (via gemeinsamem Parser).</summary>
    public required ZentralrufReplyData Data { get; init; }

    /// <summary>Hinweise auf mögliche Falschzuordnungen, die der Anwalt prüfen sollte.</summary>
    public required IReadOnlyList<string> Warnings { get; init; }

    /// <summary>
    /// Der aus der Mail gewonnene Rohtext, der durch den Parser lief. Dient der
    /// Oberfläche als Original-Ansicht/Fallback, falls das automatische Mapping
    /// unvollständig ist und der Anwalt von Hand korrigieren oder neu auswerten will.
    /// </summary>
    public string? RawText { get; init; }

    /// <summary>
    /// Vollständige Pfade der Dateien, die an der Antwort hingen (§4.3). Der
    /// Versand bietet sie zum Anhängen an (§4.7); leer, wenn nichts dranhing.
    /// </summary>
    public IReadOnlyList<string> AnhangPfade { get; init; } = [];

    /// <summary>Vom Nutzer als gesehen/übernommen markiert.</summary>
    public bool Acknowledged { get; init; }

    /// <summary>
    /// True, wenn die Verknüpfung zum Vorgang nicht über die Referenz, sondern
    /// über den Fallback (Gegner-Kennzeichen + Unfalldatum) vermutet wurde —
    /// vom Anwalt noch zu bestätigen.
    /// </summary>
    public bool ZuordnungVermutet { get; init; }
}

File: Automation_App_Frontend/lib/features/mailbox/domain/entities/received_reply.dart
import 'package:automation_app/features/zentralruf_reply/domain/entities/zentralruf_reply_data.dart';
import 'package:equatable/equatable.dart';

/// Eine vom Postfach-Monitor automatisch erfasste Zentralruf-Antwort. Die
/// extrahierten Vorgangsdaten ([data]) sind dieselben wie beim manuellen
/// Einfügen — der Monitor schickt sie durch dieselbe Auswertung.
/// Spiegelt das Backend-DTO `ReceivedReplyDto`.
class ReceivedReply extends Equatable {
  /// Prozessweite Kennung des Treffers (zum Quittieren).
  final String id;

  final DateTime receivedAt;
  final String? subject;
  final String? from;

  /// True, wenn der Anwalt den Treffer als erledigt markiert hat.
  final bool acknowledged;

  final ZentralrufReplyData data;

  /// Hinweise auf mögliche Falschzuordnungen (z. B. Kennzeichen passt nicht
  /// zur Referenz, Negativ-Antwort) — identisch zur manuellen Auswertung.
  final List<String> warnings;

  /// Dateien, die an der Antwort hingen (§4.3) — beim Versand zum Anhängen
  /// angeboten. Der Anwalt zieht sie sonst im Mailprogramm von Hand aus der
  /// erhaltenen Nachricht in die ausgehende.
  final List<String> anhangPfade;

  /// Der aus der Mail gewonnene Rohtext, der durch den Parser lief. Fallback/
  /// Diagnose, falls das automatische Mapping unvollständig ist.
  final String? rawText;

  /// True, wenn das Backend den Vorgang nicht über die Referenz, sondern über
  /// den Fallback (Gegner-Kennzeichen + Unfalldatum) vermutet hat — eine vom
  /// Anwalt noch zu bestätigende Zuordnung.
  final bool zuordnungVermutet;

  const ReceivedReply({
    required this.id,
    required this.receivedAt,
    required this.subject,
    required this.from,
    required this.acknowledged,
    required this.data,
    required this.warnings,
    this.anhangPfade = const [],
    this.rawText,
    this.zuordnungVermutet = false,
  });

  factory ReceivedReply.fromJson(Map<String, dynamic> json) {
    return ReceivedReply(
      id: json['id'] as String,
      receivedAt:
          DateTime.tryParse(json['receivedAt'] as String? ?? '')?.toLocal() ??
          DateTime.now(),
      subject: json['subject'] as String?,
      from: json['from'] as String?,
      acknowledged: json['acknowledged'] as bool? ?? false,
      data: ZentralrufReplyData.fromJson(json['data'] as Map<String, dynamic>),
      warnings: List<String>.from(json['warnings'] as List? ?? const []),
      anhangPfade: List<String>.from(json['anhangPfade'] as List? ?? const []),
      rawText: json['rawText'] as String?,
      zuordnungVermutet: json['zuordnungVermutet'] as bool? ?? false,
    );
  }

  @override
  List<Object?> get props => [
    id,
    receivedAt,
    subject,
    from,
    acknowledged,
    data,
    warnings,
    anhangPfade,
    rawText,
    zuordnungVermutet,
  ];
}
